engine/src/main/java/com/datatorrent/stram/plan/logical/LogicalPlan.java

```diff
@@ -917,6 +931,11 @@ public void addInputPort(InputPort<?> portObject, Field field, InputPortFieldAnn
         metaPort.adqAnnotation = adqAnnotation;
         inPortMap.put(portObject, metaPort);
         markInputPortIfHidden(metaPort.getPortName(), metaPort, field.getDeclaringClass());
+        if (metaPort.getStreamCodec() == null) {
+          metaPort.setStreamCodec(portObject.getStreamCodec());
+        } else if (portObject.getStreamCodec() != null) {
+          LOG.info("Input port {} overwrites codec {} with {}.", metaPort, metaPort.getStreamCodec(), portObject.getStreamCodec());
```

Comment:
Shouldn't the arguments be flipped? Should the log level be debug? What is the log level in other scenarios like overriding partitioner.